frontend/src/metabase/meta/Parameter.js

```diff
@@ -21,15 +21,109 @@ import type Field from "metabase-lib/lib/metadata/Field";
 import Dimension, { FieldDimension } from "metabase-lib/lib/Dimension";
 import moment from "moment";
 import { t } from "ttag";
+import _ from "underscore";
 import * as FIELD_REF from "metabase/lib/query/field_ref";
-import { isNumericBaseType } from "metabase/lib/schema_metadata";
+import {
+  isNumericBaseType,
+  doesOperatorExist,
+  getOperatorByTypeAndName,
+  STRING,
+  NUMBER,
+  PRIMARY_KEY,
+} from "metabase/lib/schema_metadata";
+
 import Variable, { TemplateTagVariable } from "metabase-lib/lib/Variable";
 
 type DimensionFilter = (dimension: Dimension) => boolean;
 type TemplateTagFilter = (tag: TemplateTag) => boolean;
 type FieldPredicate = (field: Field) => boolean;
 type VariableFilter = (variable: Variable) => boolean;
 
+export const PARAMETER_OPERATOR_TYPES = {
+  number: [
+    {
+      operator: "=",
+      name: t`Equal to`,
+    },
+    {
+      operator: "!=",
+      name: t`Not equal to`,
+    },
+    {
+      operator: "between",
+      name: t`Between`,
+    },
+    {
+      operator: ">=",
+      name: t`Greater than or equal to`,
+    },
+    {
+      operator: "<=",
+      name: t`Less than or equal to`,
+    },
+    // {
+    //   operator: "all-options",
+    //   name: t`All options`,
+    //   description: t`Contains all of the above`,
+    // },
+  ],
+  string: [
+    {
+      operator: "=",
+      name: t`Dropdown`,
+      description: t`Select one or more values from a list or search box.`,
+    },
+    {
+      operator: "!=",
+      name: t`Is not`,
+      description: t`Exclude one or more specific values.`,
+    },
+    {
+      operator: "contains",
+      name: t`Contains`,
+      description: t`Match values that contain the entered text.`,
+    },
+    {
+      operator: "does-not-contain",
+      name: t`Does not contain`,
+      description: t`Filter out values that contain the entered text.`,
+    },
+    {
+      operator: "starts-with",
+      name: t`Starts with`,
+      description: t`Match values that begin with the entered text.`,
+    },
+    {
+      operator: "ends-with",
+      name: t`Ends with`,
+      description: t`Match values that end with the entered text.`,
+    },
+    // {
+    //   operator: "all-options",
+    //   name: t`All options`,
+    //   description: t`Users can pick from any of the above`,
+    // },
+  ],
```

Comment:
I think this sort of stuff should go in the shared backend/frontend Clojure/ClojureScript lib in the future... we don't really need to list all the different filter types in both the FE and backend if we can just do it in one place

Reply:
Definitely agree